### PR TITLE
feat: loader for a sign request

### PR DESCRIPTION
**Pull request by jakubuid**
- Add loader for a sign request
- Add 60s timeout for subscribe, batch subscribe and unsubscribe
- Add unit tests for BaseRelayClient
- Fix crash on subscribe when proposing a session

### CODE REVIEW

**Review by TalhaAli00 — APPROVED** [review]

**Comment by jakubuid** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=WalletConnect_WalletConnectKotlinV2&pullRequest=1373) **Quality Gate failed**
> Failed conditions  [0.0% Coverage on New Code](https://sonarcloud.io/component_measures?id=WalletConnect_WalletConnectKotlinV2&pullRequest=1373&metric=new_coverage&view=list) (required ≥ 51%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=WalletConnect_WalletConnectKotlinV2&pullRequest=1373)

Tests are added, the SonarCloud is not configured properly for Foundation module

File: product/walletconnectmodal/src/main/kotlin/com/walletconnect/wcmodal/ui/WalletConnectModalViewModel.kt [hunk 1]
```diff
@@ -52,8 +46,13 @@ internal class WalletConnectModalViewModel : ViewModel() {
     }
 
     fun connect(onSuccess: (String) -> Unit) {
-        val sessionParams = WalletConnectModal.sessionParams
         try {
+            val pairing =
+                CoreClient.Pairing.create { error ->
+                    handleError(IllegalStateException("Creating Pairing failed: ${error.throwable.stackTraceToString()}"))
```
Comment (TalhaAli00) [inline product/walletconnectmodal/src/main/kotlin/com/walletconnect/wcmodal/ui/WalletConnectModalViewModel.kt:52]: we should do a return on the `connect` function if there's an error. No point in continuing

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/ParingController.kt [hunk 2]
```diff
@@ -56,5 +60,7 @@ internal class PairingControllerImpl : ParingController {
         get() = pairing.uri
 
     private fun generatePairing(): Core.Model.Pairing =
-        CoreClient.Pairing.create { error -> throw IllegalStateException("Creating Pairing failed: ${error.throwable.stackTraceToString()}") }!!.also { _pairing = it }
+        CoreClient.Pairing
+            .create { error -> logger.error(IllegalStateException("Creating Pairing failed: ${error.throwable.message}")) }!!
```
Comment (TalhaAli00) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/ParingController.kt:64]: same comment about returning on error

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/calls/ProposeSessionUseCase.kt [hunk 3]
```diff
@@ -58,7 +58,7 @@ internal class ProposeSessionUseCase(
                 val request = SignRpc.SessionPropose(params = sessionProposal)
                 proposalStorageRepository.insertProposal(sessionProposal.toVO(pairing.topic, request.id))
                 val irnParams = IrnParams(Tags.SESSION_PROPOSE, Ttl(fiveMinutesInSeconds), true)
-                jsonRpcInteractor.subscribe(pairing.topic) { error -> return@subscribe onFailure(error) }
```
Comment (TalhaAli00) [inline protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/calls/ProposeSessionUseCase.kt:61]: why do we not want to return?

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/calls/SessionAuthenticateUseCase.kt [hunk 4]
```diff
@@ -102,21 +103,23 @@ internal class SessionAuthenticateUseCase(
             },
             onFailure = { error ->
                 logger.error("Session authenticate subscribing on topic error: $responseTopic, $error")
-                return@subscribe onFailure(error)
+                onFailure(error)
```
Comment (TalhaAli00) [inline protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/calls/SessionAuthenticateUseCase.kt:106]: why not return?

File: product/walletconnectmodal/src/main/kotlin/com/walletconnect/wcmodal/ui/WalletConnectModalViewModel.kt (hunk 1 above)
Comment (jakubuid) [reply product/walletconnectmodal/src/main/kotlin/com/walletconnect/wcmodal/ui/WalletConnectModalViewModel.kt:52]: The `connect` function doesn't return anything. Do you mean adding an onError callback?

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/ParingController.kt (hunk 2 above)
Comment (jakubuid) [reply product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/ParingController.kt:64]: Actually, reverted this, it was handled fine

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/calls/ProposeSessionUseCase.kt (hunk 3 above)
Comment (jakubuid) [reply protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/calls/ProposeSessionUseCase.kt:61]: This return doesn't do anything. Even if an error on subscribe is returned the code is executing and the proposal is sent.

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/calls/SessionAuthenticateUseCase.kt (hunk 4 above)
Comment (jakubuid) [reply protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/calls/SessionAuthenticateUseCase.kt:106]: This return doesn't do anything. Even if an error on subscribe is returned the code is executing and the proposal is sent.

**Pull request merged**